### PR TITLE
Elasticsearch: Add test and documentation for secured cluster, bump default Elasticsearch version (deprecated constructor) from 6.4.1 to 7.9.2

### PR DESCRIPTION
**Pull request by dadoonet**
This can be now activated for free with the default basic license.

### CODE REVIEW

File: modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java [hunk 1]
```diff
@@ -34,15 +34,15 @@
     /**
      * Elasticsearch Default version
      */
-    protected static final String ELASTICSEARCH_DEFAULT_VERSION = "6.4.1";
```
Comment (rnorth) [inline modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java]: Unfortunately I think we have to be careful about this; anybody using the plain `ElasticsearchContainer()` constructor in their tests will get an unexpected major version bump.

We have a similar situation with other modules where the default version is getting old, and I would like to advance versions - but don't want to cause unexpected breakage.

@bsideup, @kiview let's discuss the most responsible course of action...

Comment (dadoonet) [reply modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java]: Fortunately the documentation invites the user to use a specific version.
As we did not document the default constructor (which we should may be ban) I'd not be super concerned by this breaking change.

My 2 cents.

**Comment by rnorth** [comment]
Not stale. Sorry for being slow to come back to this PR.

**Comment by rnorth** [comment]
With #2839 we're intending to _deprecate_ the default constructor for all modules and start encouraging all users to specify a version explicitly. This is primarily to avoid the risk of us breaking users' builds by changing the docker image unexpectedly.

As such, I'll close this PR, as I'd expect that we will leave default image tags where they are until they eventually get removed altogether.

**Comment by dadoonet** [comment]
@rnorth I disagree because the change I proposed is bringing much more value than just updating the default version.
It brings new tests, new documentation, etc...

So please consider reopening it.

I 100% agree of removing the default constructor though but this is another story IMO.

File: modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java [hunk 2]
```diff
@@ -34,15 +34,15 @@
...
+    protected static final String ELASTICSEARCH_DEFAULT_VERSION = "7.5.2";
 
     public ElasticsearchContainer() {
```
Comment (dadoonet) [inline modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java:39]: Let's just mark this as `@Deprecated`

File: modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java [hunk 3]
```diff
@@ -67,14 +77,34 @@ public void elasticsearchDefaultTest() throws IOException {
         }
     }
 
+    @Test
+    public void elasticsearchSecuredTest() throws IOException {
+        try (ElasticsearchContainer container = new ElasticsearchContainer()
```
Comment (dadoonet) [inline modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java:82]: We should just replace this default constructor method

**Comment by rnorth** [comment]
@dadoonet, sorry, I missed your reply. I realise now what you mean - yes, we should bring the other elements of this PR in.

Can we refactor this once #2839 is merged, so that the docs refer to the right (new) way of doing things? We should still leave the default version unchanged and deprecated, but there's absolutely nothing to stop the tests and docs referring to newer versions.

Sorry for my misunderstanding.

**Comment by dadoonet** [comment]
Thanks! I subscribed to the PR and will update the PR when merged. Although I could have done it before your merge. But no worries.

**Comment by dadoonet** [comment]
@rnorth I revisited the PR and removed all calls to the default CTOR and mark it as Deprecated so I think it's aligned with what you want to merge in master with #2839. 

I'd be happy to hear your feedback.

**Comment by dadoonet** [comment]
I rebased my work on master. Could you give another look?

**Comment by dadoonet** [comment]
I believe the branch is now ready for review as CI is happy :)

**Review by rnorth — APPROVED** [review]
@dadoonet this looks great, thank you. Thanks for your patience on bumping the version of the docker image as well - I'm very happy with how this has turned out!

**Pull request merged**